todo task for gci and bug fix

samsruti opened the pull request:
Link to the Todo : https://github.com/flavour/eden/blob/master/modules/s3db/deploy.py#L865
GCI task link : http://www.google-melange.com/gci/task/view/google/gci2014/5292591847309312

In this line : https://github.com/edengci/eden/blob/staging/modules/s3db/event.py#L2346 , settings = current.deployment_settings was missing .
So when we try to update an event details ( like : http://127.0.0.1:8000/eden/event/event/1/update)
when we click on the 'Send Notification tab' as ticket error said global name 'settings' is not defined.
So I added  "settings = current.deployment_settings" in L2285 , so that "send notification" tab can work properly.

hitesh96db commented on modules/s3db/deploy.py:
```diff
@@ -862,8 +862,16 @@ def model(self):
-                                    # @todo:
-                                    #S3SQLInlineComponent("document"),
+                                    S3SQLInlineComponent(
+                                        "document",
```
Won't work because document isn't added as a component yet. Look at other places where document has been used as a component.

samsruti replied:
Got it. I am on the way to fix that :+1:

hitesh96db commented on modules/s3db/deploy.py:
```diff
@@ -875,6 +879,33 @@ def model(self):
+                                     )        
+
+        tablename = "deploy_response_document"
+        define_table(tablename,
```
I don't think there's a need for a table for documents. Documents is used as a component in a lot of places, and can be used as a super entity. 
eg: https://github.com/flavour/eden/blob/master/private/templates/SSF/config.py#L342
https://github.com/flavour/eden/blob/master/modules/s3db/project.py#L155
What you've done is for defining components for a specific resource, that might not be used in several other places..
Look here as well http://eden.sahanafoundation.org/wiki/S3/S3Model/SuperEntities

samsruti committed:
ToDo task for GCI ,bug fix: "global name 'settings' is not defined "error
is fixed

ptressel commented:
It's good to use a commit message that says what the fix is for.  You can include a link to the ticket or the todo.  So, could you change the commit message to say what the two changes do?  There is a special git command just for that, since it happens all the time:

git commit --amend

Do this in the branch that has your work as the top commit, and nothing in the staging area -- then it will just bring up the commit message editor so you can change it.  (I don't know where this is in the Git for Windows or Git Gui menus...)  Here's a description:

http://git-scm.com/book/en/v2/Git-Basics-Undoing-Things

samsruti commented:
:+1:  Commit message changed. In the decription , i explained how the error generated and how I fixed that.
I also added the link to the todo and to the GCI task.

ramdesh commented:
Merged from @samsruti for GCI Task http://www.google-melange.com/gci/task/view/google/gci2014/5292591847309312
